### PR TITLE
adding in handling for ssl_dhparam

### PR DESCRIPTION
**Pull request by nellshamrell**
Fix for https://github.com/chef/supermarket/issues/1057

Enhances security for dbparam, see https://github.com/chef/chef-server/pull/301 for reference

### CODE REVIEW

**Comment by nellshamrell** [comment]
I would love advice on testing this.  I can build the package (which is a time intensive process, but doable) and install it on a test kitchen instance (or live VM), but am interested in whether there's a way to test this locally.

Thanks!

File: cookbooks/omnibus-supermarket/recipes/ssl.rb [hunk 1]
```diff
@@ -71,8 +72,19 @@
     end
   end
 
+  if ! File.exists?(ssl_dhparam)
```
Comment (smith) [inline cookbooks/omnibus-supermarket/recipes/ssl.rb]: Since you're doing `:create_if_missing`, you don't need to wrap this in an if statement.

**Comment by smith** [comment]
You could use the test kitchen setup in the cookbook: https://github.com/chef/omnibus-supermarket/tree/0d1d53ffd410ad7ce6215d9eb53cf8dc0e1c01d2/cookbooks/omnibus-supermarket#integration

There are some tests in there already: https://github.com/chef/omnibus-supermarket/blob/0d1d53ffd410ad7ce6215d9eb53cf8dc0e1c01d2/cookbooks/omnibus-supermarket/test/integration/default/serverspec/recipes/nginx_spec.rb, you could add a check to see if that file has been created and if it looks like you think it should.

**Comment by nellshamrell** [comment]
Had some test failures, so went back through this step by step adding in tests along with way.  Moving to this pull request https://github.com/chef/omnibus-supermarket/pull/25

**Pull request closed without merging**